CMakeLists.txt

```diff
@@ -12,7 +12,8 @@ cmake_minimum_required(VERSION 3.21)
 # Additional include directories for CMake utils.
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/cmake)
 
-# TODO(DF): Set CMAKE_OSX_DEPLOYMENT_TARGET when it's time to tackle OSX
+# Minimum required for the C++17 features used.
+set(CMAKE_OSX_DEPLOYMENT_TARGET 10.15)
```

Comment:
[question] Isn't this the same variable as `MACOSX_DEPLOYMENT_TARGET` set in `build-wheels`? Does it need to be set twice?

Reply:
It needs to be set for any build in CMake to ensure that the Clang is set to the right macOS SDK version (otherwise it errors as we're using features in `std::filesystem` that are only available in macOS 10.15 or later). It need to be set for `cibuildwheel` so that it sets the correct metadata in the wheel so it won't install on systems earlier than 10.15 - as that process is the parent environment for CMake so it can't see the vars CMake sets.

Reply:
:+1: